cmd/czds-dl/download.go

```diff
@@ -103,22 +110,32 @@ func main() {
 func addLinks(downloads []string) {
 	for _, dl := range downloads {
 		work.Add(1)
-		inputChan <- dl
-	}
-	close(inputChan)
+		inputChan <- &ZoneInfo{
+			Dl:    dl,
+			Count: 1,
+		}
+	}	
 	loadDone <- true
 }
 
 func worker() {
 	for {
-		dl, more := <-inputChan
+		zi, more := <-inputChan
 		if more {
 			// do work
-			err := zoneDownload(dl)
+			err := zoneDownload(zi)
 			if err != nil {
 				// don't stop on an error that only affects a single zone
 				// fixes occasional HTTP 500s from CZDS
-				log.Print(err)
+				v("[%s] err: %s", path.Base(zi.Dl), err)
+				zi.Count++
+				if uint(zi.Count) < *retries {
+					work.Add(1)
+					inputChan <- zi // requeue
+				} else {
+					log.Printf("[%s] Max fail count hit; not downloading.", path.Base(zi.Dl))
+					_ = os.Remove(zi.FullPath)
```

Comment:
Can you add error checking here?
It does not need to be a fatal error, but printing it to stderr if something wonky is going on with the filesystem is important.